pandas/core/series.py

```diff
@@ -2623,22 +2624,57 @@ def _binop(self, other, func, level=None, fill_value=None):
         if not isinstance(other, Series):
             raise AssertionError("Other operand must be Series")
 
-        new_index = self.index
         this = self
 
         if not self.index.equals(other.index):
             this, other = self.align(other, level=level, join="outer", copy=False)
-            new_index = this.index
 
         this_vals, other_vals = ops.fill_binop(this.values, other.values, fill_value)
 
         with np.errstate(all="ignore"):
             result = func(this_vals, other_vals)
 
         name = ops.get_op_result_name(self, other)
-        ret = ops._construct_result(self, result, new_index, name)
+        ret = this._construct_result(result, name)
         return ret
 
+    def _construct_result(
```

Comment:
maybe contrust_result_from_op

Reply:
for now mirroring the naming in DataFrame